src/qibolab/serialize.py

```diff
@@ -79,7 +80,52 @@ def load_qubits(
     return qubits, couplers, pairs
 
 
-# This creates the compiler error
+def _load_pulse(pulse_kwargs, qubit):
+    pulse_type = pulse_kwargs.pop("type")
+    if "coupler" in pulse_kwargs:
+        q = pulse_kwargs.pop("coupler", qubit.name)
+    else:
+        q = pulse_kwargs.pop("qubit", qubit.name)
+
+    if pulse_type == "dl":
+        return Delay(**pulse_kwargs)
+    if pulse_type == "virtual_z":
+        return VirtualZ(**pulse_kwargs, qubit=q)
+    return Pulse(**pulse_kwargs, type=pulse_type, qubit=q)
```

Comment:
That's another good reason to avoid both alternative classes and subclasses: this could be as simple as:
```py
Pulse(**kwargs, qubit=qubit)
```
(after getting rid of `Coupler` niceties...)